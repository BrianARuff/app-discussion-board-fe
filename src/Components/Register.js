import React from "react";
import { Link } from "react-router-dom";
import { InputLabel, FormHelperText, Input, FormControl, Button } from "@material-ui/core";
import { StyledDropZone } from 'react-drop-zone'
import 'react-drop-zone/dist/styles.css';
import axios from "axios";
import Cookie from "js-cookie";

import "../styles/register.css";
export default class Register extends React.Component {
  state = {
    name: "",
    email: "",
    password: "",
    date_of_birth: "",
    description: "",
    image: "",
    role: "member",
    status: "Register"
  }

  handleInputChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  registerUser = () => {
    this.setState({status: "registering"});
    axios.post("https://suicide-watch-backend.herokuapp.com/auth/register", this.state)
      .then(res => {
        localStorage.setItem("username", res.data.user.name);
        Cookie.set("token", res.data.token, {
          expires: 7,
          path: "/"
        });
        this.setState({status: "Registered"});
        this.props.history.push('/');
      })
      .catch(err => {
        console.error(this.setState({status: "registering"}));
        this.setState({status: "Error registering account... Try again."});
      });
  }

  render() {
    return (
      <div className="register-form-container">
        <h1>Register Below</h1>
        <div className="register-form-flex">
          <FormControl fullWidth required>
            <InputLabel>
              Username
              </InputLabel>
            <Input name="name" type="text" onChange={this.handleInputChange} />
            <FormHelperText>Please enter a username that you are comfortable using and can remember</FormHelperText>
          </FormControl>

          <FormControl fullWidth required>
            <InputLabel>
              Email
            </InputLabel>
            <Input name="email" type="email" onChange={this.handleInputChange} />
            <FormHelperText>Please enter a valid email address that you can keep up with</FormHelperText>
          </FormControl>

          <FormControl fullWidth required>
            <InputLabel>
              Password
            </InputLabel>
            <Input name="password" type="password" onChange={this.handleInputChange} />
            <FormHelperText>Please enter a strong password that you can remember</FormHelperText>
          </FormControl>

          <FormControl fullWidth >
            <InputLabel />
            <Input name="date_of_birth" type="date" onChange={this.handleInputChange} />
            <FormHelperText>Please enter a valid email address that you can keep up with</FormHelperText>
          </FormControl>

          <FormControl fullWidth>
            <InputLabel>
              Description
            </InputLabel>
            <Input name="description" type="text" onChange={this.handleInputChange} />
            <FormHelperText>Please give us a short description about yourself, if you want.</FormHelperText>
          </FormControl>

          <br />
          <StyledDropZone label="Select or Drop your image here" onDrop={(file, text) => {
            const reader = new FileReader();
            reader.onload = upload => {
              this.setState({ image: upload.srcElement.result });
            };
            reader.readAsDataURL(file);
          }}>
          </StyledDropZone>

          <br />
          <FormControl fullWidth>
            <Button
              onClick={this.registerUser}
              fullWidth
              color="secondary"
              variant="contained"
            >
              {this.state.status}
            </Button>
            <FormHelperText>Already registered? Please go <Link to="/login">Login</Link> to login.</FormHelperText>
          </FormControl>
        </div>
      </div>
    )
  }
}