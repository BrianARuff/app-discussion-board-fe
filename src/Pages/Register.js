import React from "react";
import { Link } from "react-router-dom";
import { InputLabel, FormHelperText, Input, FormControl, Button } from "@material-ui/core";
import { StyledDropZone } from 'react-drop-zone'
import LoopIcon from '@material-ui/icons/Loop';
import 'react-drop-zone/dist/styles.css';
import axios from "axios";
import Cookie from "js-cookie";

import "../styles/register.css";
export default class Register extends React.Component {
  state = {
    name: "",
    email: "",
    password: "",
    date_of_birth: "",
    description: "",
    image: "",
    role: "member",
    status: "Register",
    dropZoneText: "Select or Drop your Profile Image Here"
  }

  spinnerRef = React.createRef();

  componentDidMount() {
    this.spinnerRef.current.className = "hide";
  }

  handleInputChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  registerUser = () => {
    this.spinnerRef.current.className = "spin";
    this.setState({ status: "Registering" });
    axios.post("https://suicide-watch-backend.herokuapp.com/auth/register", this.state)
      .then(res => {
        Cookie.set("token", res.data.token, {
          expires: 7,
          path: "/"
        });
        this.setState({ status: "Registered" });
        this.props.history.push('/');
        window.location.reload();
        window.scrollTo({
          top: 0,
          behavior: "auto"
        });
        this.spinnerRef.current.className = "hide";
      })
      .catch(err => {
        this.setState({ status: "Error registering account... Try again." });
        this.spinnerRef.current.className = "hide";
      });
  }

  imageUploadProgress = e => {
    let progress = Math.round(e.loaded / e.total) * 100;
    if (progress < 100) {
      this.setState({dropZoneText: (progress + "%")});
    } else {
      this.setState({dropZoneText: "Your Profile Image was Uploaded"});
      this.spinnerRef.current.className = ""
    }
  }

  registerUserEnter = (e) => e.key === "Enter" ? this.registerUser() : null;

  render() {
    return (
      <div className="register-form-container">
        <h1 className="hide">Register Page</h1>
        <h2 className="fw-bold">Register Below</h2>
        <form onKeyDown={(e) => this.registerUserEnter(e)} className="register-form-flex">
          <FormControl fullWidth required>
            <InputLabel>
              Username
              </InputLabel>
            <Input autoComplete="true" name="name" type="text" onChange={this.handleInputChange} />
            <FormHelperText>Please enter your username</FormHelperText>
          </FormControl>

          <FormControl fullWidth required>
            <InputLabel>
              Email
            </InputLabel>
            <Input autoComplete="true" name="email" type="email" onChange={this.handleInputChange} />
            <FormHelperText>Please enter a valid email address that you can keep up with</FormHelperText>
          </FormControl>

          <FormControl fullWidth required>
            <InputLabel>
              Password
            </InputLabel>
            <Input autoComplete="true" name="password" type="password" onChange={this.handleInputChange} />
            <FormHelperText>Please enter a strong password that you can remember</FormHelperText>
          </FormControl>

          <FormControl fullWidth >
            <InputLabel />
            <Input name="date_of_birth" type="date" onChange={this.handleInputChange} />
            <FormHelperText>Please enter a valid email address that you can keep up with</FormHelperText>
          </FormControl>

          <FormControl fullWidth>
            <InputLabel>
              Description
            </InputLabel>
            <Input name="description" type="text" onChange={this.handleInputChange} />
            <FormHelperText>Please give us a short description about yourself, if you want.</FormHelperText>
          </FormControl>

          <br />
          <FormControl fullWidth>
            <StyledDropZone 
              label={this.state.dropZoneText}
              onDrop={(file, text) => {
                const reader = new FileReader();
                reader.onload = upload =>
                this.setState({ image: upload.srcElement.result });
                reader.onprogress = this.imageUploadProgress
                reader.readAsDataURL(file);
              }}>
            </StyledDropZone>
          </FormControl>

          <br />
          <FormControl fullWidth>
            <Button
              onClick={this.registerUser}
              fullWidth
              color="secondary"
              variant="contained"
            >
              <span style={{position: "relative", left: -10, paddingTop: "5px"}} ref={this.spinnerRef} className={this.state.spinner}><LoopIcon /></span>
              {this.state.status}
            </Button>
            <FormHelperText>Already registered? Please go <Link to="/login">Login</Link> to login.</FormHelperText>
          </FormControl>
        </form>
      </div>
    )
  }
}