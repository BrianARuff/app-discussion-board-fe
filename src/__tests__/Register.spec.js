import React from "react";
import { create } from "react-test-renderer";
import Register from "../Pages/Register.js";

describe("test register component", () => {
  test("for snapshot testing", () => {
    const register = create(<Register />);
    expect(register.toJSON()).toMatchSnapshot();
  });
});
