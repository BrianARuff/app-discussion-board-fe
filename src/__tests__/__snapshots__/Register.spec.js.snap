// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test register component for snapshot testing 1`] = `
<div
  className="register-form-container"
>
  <h1
    className="hide"
  >
    Register Page
  </h1>
  <h2
    className="fw-bold"
  >
    Register Below
  </h2>
  <form
    className="register-form-flex"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
      >
        Username
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="true"
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
      </div>
      <p
        className="MuiFormHelperText-root Mui-required"
      >
        Please enter your username
      </p>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
      >
        Email
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="true"
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
        />
      </div>
      <p
        className="MuiFormHelperText-root Mui-required"
      >
        Please enter a valid email address that you can keep up with
      </p>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
      >
        Password
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="true"
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
        />
      </div>
      <p
        className="MuiFormHelperText-root Mui-required"
      >
        Please enter a strong password that you can remember
      </p>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      />
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="date_of_birth"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="date"
        />
      </div>
      <p
        className="MuiFormHelperText-root"
      >
        Please enter a valid email address that you can keep up with
      </p>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        Description
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
      </div>
      <p
        className="MuiFormHelperText-root"
      >
        Please give us a short description about yourself, if you want.
      </p>
    </div>
    <br />
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <div
        className="DropZone"
        onClick={[Function]}
        onDrag={[Function]}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        role="button"
      >
        Select or Drop your Profile Image Here
      </div>
    </div>
    <br />
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Register
        </span>
      </button>
      <p
        className="MuiFormHelperText-root"
      >
        Already registered? Please go
         
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
         to login.
      </p>
    </div>
  </form>
</div>
`;
